File: Java/3_TwoPointers/LC_2562_FindTheArrayConcatenationValue.java
class Solution {
    public long findTheArrayConcVal(int[] nums) {
        int left = 0;
        int right = nums.length - 1;
        long sum = 0;
        StringBuilder sb;
        while (left < right) {
            sb = new StringBuilder();
            sb.append(nums[left++]);
            sb.append(nums[right--]);
            sum += Integer.parseInt(sb.toString());
        }

        sum += left == right ? nums[left] : 0;
        return sum;
    }
}

public class LC_2562_FindTheArrayConcatenationValue {

}

File: Swift/Leetcode.playground/Sources/3_TwoPointers/LC_2562_FindTheArrayConcatenationValue.swift
class Solution {
    func findTheArrayConcVal(_ nums: [Int]) -> Int {
        var left = 0
        var right = nums.count - 1
        var sum = 0
        var sb: String
        while left < right {
            sb = ""
            sb.append(String(nums[left]))
            sb.append(String(nums[right]))
            sum += Int(sb)!
            left += 1
            right -= 1
        }
        sum += left == right ? nums[left] : 0
        return sum
    }
}